addons/website/static/src/js/menu/content.js

```diff
@@ -879,6 +879,7 @@ var ContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
         return new Promise(function (resolve) {
             var dialog = new EditMenuDialog(self, {}, rootID);
             dialog.on('save', self, function () {
+                window.document.body.classList.add('o_wait_reload');
```

Comment:
I think it is not clean as well but I did not find a way to wait for the page to be reloaded after saving the link from the dialog...
As there is a delay between the refresh and the time that the modal is closed, for a test that is run without any delay the extraMenuItem dropdown-toggle will be clicked before the page is reloaded and the menu added. So I went for the use of the existing class `o_wait_reload` but maybe it would be cleaner to use window.onload inside the tests ?